diff -ruN Net-DNS-0.83.orig/lib/Net/DNS/Resolver/UNIX.pm Net-DNS-0.83/lib/Net/DNS/Resolver/UNIX.pm
--- Net-DNS-0.83.orig/lib/Net/DNS/Resolver/UNIX.pm	2015-02-26 16:01:55.000000000 +0100
+++ Net-DNS-0.83/lib/Net/DNS/Resolver/UNIX.pm	2015-03-06 12:37:21.217583476 +0100
@@ -32,7 +32,17 @@
 sub init {
 	my $defaults = shift->defaults;
 
+	eval {
 	$defaults->read_config_file($resolv_conf) if -f $resolv_conf && -r _;
+	};
+	if ( $@ ) {
+		if ( $@ =~ /^Could not open / ) {
+			# we ignore this problem
+			#warn "$@";
+		} else {
+			die $@;
+		}
+	}
 
 	$defaults->domain( _untaint $defaults->domain );	# untaint config values
 	$defaults->searchlist( _untaint $defaults->searchlist );
